src/tritongrader/test_case/io_test_case.py

```diff
@@ -133,7 +129,17 @@ def execute(self):
                 arm=self.arm,
             )
             self.runner.run()
-            self.result.passed = self.check_output()
+            stdout_check = self.runner.check_stdout(self.exp_stdout_path)
+            stderr_check = self.runner.check_stderr(self.exp_stderr_path)
+            status = True
+            if self.exp_exit_status is not None:
+                status = self.exp_exit_status == self.runner.exit_status
+            self.result.passed = stdout_check and stderr_check and status
+            self.exit_status: int = self.runner.exit_status
+            print(
+                f"stdout check: {stdout_check}; stderr check: {stderr_check}; status: {status}"
+            )
```

Comment:
Let's stick all of this into a function called `check_test_passed` or something. Would make the code cleaner.

Reply:
Refactored this a little, don't think the new version would require another function, but feel free to raise again in new PR